python/vsi/test/test_file_utils.py

```diff
@@ -0,0 +1,137 @@
+         with open(foo_file,'w') as foo:
+           foo.write('test')
+         # create a read only file to test the ignore_errors variable
+         if key == 'sub_dir2':
+           os.chmod(foo_file, S_IREAD)
+
+     # Check that the temp directory exists.
+     if not os.path.isdir(foo_dir):
+       raise ValueError (
+         '{} does not exist! It is a temporary directory that is deleted once it is out of scope.'
+         .format(foo_dir))
+     elif os.path.isdir(foo_dir):
+       # remove directory recursively but keep the top dir
+       file_utils.rmtree(foo_dir, ignore_errors=False, rmdir=False)
+       if not os.path.isdir(foo_dir):
+         raise ValueError ('{} should still exist since rmdir is False.'.format(foo_dir))
+       # now remove the top dir
+       elif os.path.isdir(foo_dir):
+         file_utils.rmtree(foo_dir)
+       # This directory should no longer exist.
+       if os.path.isdir(foo_dir):
+         raise ValueError ('{} should no longer exist.'.format(foo_dir))
+
+
+  def test_get_files_with_extension_from_dir(self):
+
+    # tests finding files with the specified extension(s)
+    # first create a temp directory
+    with tempfile.TemporaryDirectory() as foo_dir:
+
+      # temp files
+      foo_files = []
+      # an empty list is returned if no files with the given extension exist.
+      extensions = ['tif', 'tiff']
+      files = file_utils.get_files_with_extension_from_dir(foo_dir, extensions)
+      self.assertEqual(files, foo_files)
+
+      for key, value in sub_foo_dirs.items():
+        sub_foo_dir = os.path.join(foo_dir,key)
+
+        # Make the subdirectory
+        os.makedirs(sub_foo_dir)
+        # create some files in the subdirectory
+        for f in value:
+          foo_file = os.path.join(sub_foo_dir, f)
+          # write to each file
+          with open(foo_file,'w') as foo:
+            foo.write('test')
+          foo_files.append(foo_file)
+
+      # sort the list of foo files
+      foo_files.sort()
+      # test finding the files with the specified extension(s)
+      extensions = ['txt']
+      files = file_utils.get_files_with_extension_from_dir(foo_dir, extensions)
+      # sort the list of expected files
+      files.sort()
+      self.assertEqual(files, foo_files)
+
+
+  def test_get_neighbor_files_with_extension(self):
+
+    # tests finding files in the same directory,
+    # with the same name, but have different extensions
+    # first create a subdirectory
+    with tempfile.TemporaryDirectory() as foo_dir:
+
+      # create some foo files with the same name, but different extensions
+      # a text file
+      foo_text_file = os.path.join(foo_dir, 'temp1.txt')
+      with open(foo_text_file, 'w') as foo:
+        foo.write('test')
+
+      # an image file
+      foo_image_file = os.path.join(foo_dir, 'temp1.tif')
+      foo_data = np.random.randint(0, 255, (256, 256, 3), 'uint8')
+      imwrite(foo_image_file, foo_data, photometric='rgb')
```

Comment:
Does this need to be a real tiff file for this test? Can't it just be an empty file?